controller/users.go

```diff
@@ -166,6 +166,11 @@ func (c *UsersController) Update(ctx *app.UpdateUsersContext) error {
 
 		updatedEmail := ctx.Payload.Data.Attributes.Email
 		if updatedEmail != nil && *updatedEmail != user.Email {
+			isValid := isUsernameValid(*updatedEmail)
+			if !isValid {
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(goa.ErrInvalidRequest(fmt.Sprintf("invalid value assigned to email for identity with id %s and user with id %s", identity.ID, identity.UserID.UUID)))
```

Comment:
We should probably add a `NewBadRequestError(msg string)` method in `errors/errors.go` along with a `BadRequestError` structure and then call it like that:
````
return jsonapi.JSONErrorResponse(ctx, errors.NewBadRequestError(fmt.Sprintf("invalid value assigned to email for identity with id %s and user with id %s", identity.ID, identity.UserID.UUID)))
````